content/webapp/components/RelatedWorks/RelatedWorks.helpers.tsx

```diff
@@ -0,0 +1,138 @@
+import { toHtmlId } from '@weco/common/utils/grammar';
+import { catalogueQuery } from '@weco/content/services/wellcome/catalogue';
+import {
+  toWorkBasic,
+  Work,
+  WorkBasic,
+} from '@weco/content/services/wellcome/catalogue/types';
+import { Toggles } from '@weco/toggles';
+
+// Returns the century range for a string containing exactly four digits
+const getCenturyRange = (
+  str?: string
+): { tabLabel: string; from: string; to: string } | null => {
+  const match = str?.match(/^(\d{4})$/);
+
+  if (match) {
+    const year = parseInt(match[0], 10);
+    const centuryStart = Math.floor(year / 100) * 100;
+    const centuryEnd = centuryStart + 99;
+
+    return {
+      tabLabel: `From ${centuryStart}s`,
+      from: `${centuryStart}-01-01`,
+      to: `${centuryEnd}-12-31`,
+    };
+  }
+  return null;
+};
+
+export const fetchRelatedWorks = async ({
+  work,
+  toggles,
+  setIsLoading,
+}: {
+  work: Work;
+  toggles: Toggles;
+  setIsLoading: (isLoading: boolean) => void;
+}): Promise<{
+  [key: string]: { label: string; results: WorkBasic[] };
+}> => {
+  setIsLoading(true);
+  const results: {
+    [key: string]: { label: string; results: WorkBasic[] };
+  } = {};
+
+  const subjectLabels = work.subjects.map(subject => subject.label).slice(0, 3);
+  const typeTechniques = work.genres.map(genres => genres.label).slice(0, 3);
+  const dateRange = getCenturyRange(work.production[0]?.dates[0]?.label);
+
+  const catalogueBasicQuery = async params =>
+    await catalogueQuery('works', {
+      toggles,
+      // Always fetch 4 works in case we get the current work back, then we will still have 3 to show.
+      pageSize: 4,
+      params: {
+        include: ['production', 'contributors'], // This returns minimal data
+        ...params,
+      },
+    });
+
+  const addToResultsObject = (
+    categoryLabel: string,
+    tabLabel: string,
+    response
```

Comment:
Should we try to type this (my editor has put a squiggly line under it)?

Reply:
Yes can do! Weird that I didn't get the flag? 🤔